### PR TITLE
pip installation via pyproject

### PR DESCRIPTION
**Pull request by hardikdava**
@yzslab I have added support for pip installation. It can be used as

`pip install git+https://github.com/yzslab/gaussian-splatting-lightning.git`

### CODE REVIEW

**Comment by hardikdava** [comment]
Related to #50

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    return install_requires, dependency_links
+
+with open("./requirements.txt", "r") as f:
+    requirements = f.read().splitlines()
+
+install_requires, dependency_links = split_requirements(requirements)
+
+setup(
+    name = "gsplat_lightning",
```
Comment (yzslab) [inline setup.py:27]: I prefer not to contain `gsplat`. How about the full name of this repo?

**Review by yzslab — CHANGES_REQUESTED** [review]

File: requirements.txt [hunk 2]
```diff
@@ -9,5 +9,5 @@ viser==0.2.3
 opencv-python==4.10.*
 matplotlib
 mediapy==1.2.2
-./submodules/diff-gaussian-rasterization
-./submodules/simple-knn
\ No newline at end of file
+git+https://github.com/graphdeco-inria/diff-gaussian-rasterization.git@59f5f77e3ddbac3ed9db93ec2cfe99ed6c5d121d
+git+https://gitlab.inria.fr/bkerbl/simple-knn
```
[the remark this answers is not in the document]
Comment (yzslab) [reply requirements.txt:13]: If install from local is impossible, remember to remove them from git submodule.
If they have to be install from url, shall here also specify the hash for simple-knn?

[the remark this answers is not in the document]
Comment (hardikdava) [reply requirements.txt]: @yzslab I tried some options but could not able to install sub-module through local directory.

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            dependency_links.append(requirement)
+        else:
+            install_requires.append(requirement)
+
+    return install_requires, dependency_links
+
+with open("./requirements.txt", "r") as f:
+    requirements = f.read().splitlines()
+
+install_requires, dependency_links = split_requirements(requirements)
```
[the remark this answers is not in the document]
Comment (hardikdava) [reply setup.py]: updated the version

**Comment by hardikdava** [comment]
@yzslab here is the logs of installation. It successfully installed all packages.

```
Requirement already satisfied: jmespath<2.0.0,>=0.7.1 in /home/hd/Desktop/python_workspace/supervision/venv/lib/python3.10/site-packages (from boto3->lightning-cloud>=0.5.38->lightning[pytorch-extra]==2.0.9.post0->gaussian-splatting-lightning==0.1.0) (1.0.1)
Requirement already satisfied: s3transfer<0.11.0,>=0.10.0 in /home/hd/Desktop/python_workspace/supervision/venv/lib/python3.10/site-packages (from boto3->lightning-cloud>=0.5.38->lightning[pytorch-extra]==2.0.9.post0->gaussian-splatting-lightning==0.1.0) (0.10.2)
Requirement already satisfied: PySocks!=1.5.7,>=1.5.6 in /home/hd/Desktop/python_workspace/supervision/venv/lib/python3.10/site-packages (from requests<4.0->lightning[pytorch-extra]==2.0.9.post0->gaussian-splatting-lightning==0.1.0) (1.7.1)
Requirement already satisfied: executing>=1.2.0 in /home/hd/Desktop/python_workspace/supervision/venv/lib/python3.10/site-packages (from stack-data->ipython->mediapy==1.2.2->gaussian-splatting-lightning==0.1.0) (2.1.0)
Requirement already satisfied: asttokens>=2.1.0 in /home/hd/Desktop/python_workspace/supervision/venv/lib/python3.10/site-packages (from stack-data->ipython->mediapy==1.2.2->gaussian-splatting-lightning==0.1.0) (2.4.1)
Requirement already satisfied: pure-eval in /home/hd/Desktop/python_workspace/supervision/venv/lib/python3.10/site-packages (from stack-data->ipython->mediapy==1.2.2->gaussian-splatting-lightning==0.1.0) (0.2.3)
Requirement already satisfied: mpmath>=0.19 in /home/hd/Desktop/python_workspace/supervision/venv/lib/python3.10/site-packages (from sympy->torch<4.0,>=1.11.0->lightning[pytorch-extra]==2.0.9.post0->gaussian-splatting-lightning==0.1.0) (1.3.0)
Requirement already satisfied: httpcore==1.* in /home/hd/Desktop/python_workspace/supervision/venv/lib/python3.10/site-packages (from httpx->trimesh>=3.21.7->viser==0.2.3->gaussian-splatting-lightning==0.1.0) (1.0.5)
Requirement already satisfied: jsonschema-specifications>=2023.03.6 in /home/hd/Desktop/python_workspace/supervision/venv/lib/python3.10/site-packages (from jsonschema->trimesh>=3.21.7->viser==0.2.3->gaussian-splatting-lightning==0.1.0) (2023.12.1)
Requirement already satisfied: referencing>=0.28.4 in /home/hd/Desktop/python_workspace/supervision/venv/lib/python3.10/site-packages (from jsonschema->trimesh>=3.21.7->viser==0.2.3->gaussian-splatting-lightning==0.1.0) (0.35.1)
Requirement already satisfied: rpds-py>=0.7.1 in /home/hd/Desktop/python_workspace/supervision/venv/lib/python3.10/site-packages (from jsonschema->trimesh>=3.21.7->viser==0.2.3->gaussian-splatting-lightning==0.1.0) (0.20.0)
Building wheels for collected packages: gaussian-splatting-lightning
  Building wheel for gaussian-splatting-lightning (pyproject.toml) ... done
  Created wheel for gaussian-splatting-lightning: filename=gaussian_splatting_lightning-0.1.0-py3-none-any.whl size=219875 sha256=cb2c5aec794244051a58ba01ab2b7ee30e8b25b241ecc0321314e93450dc9edf
  Stored in directory: /tmp/pip-ephem-wheel-cache-g3yzy5a8/wheels/72/df/8f/b0e27189bb364b389b0a8c30d8ad200b69bc84dc591e4835dd
Successfully built gaussian-splatting-lightning
Installing collected packages: simple-knn, diff-gaussian-rasterization, gaussian-splatting-lightning
Successfully installed diff-gaussian-rasterization-0.0.0 gaussian-splatting-lightning-0.1.0 simple-knn-0.0.0

```

**Comment by hardikdava** [comment]
@yzslab now, we can also remove `requirements.txt` along with some submodules.

**Comment by yzslab** [comment]
Hi, take a look whether my updates work for you.

**Comment by hardikdava** [comment]
@yzslab I will test it out and let you know. Thanks

**Comment by hardikdava** [comment]
@yzslab I have tested the installation and it is working fine. It is also installing both `diff-gaussian-rasterization` and `simple-knn` as default option.

**Comment by hardikdava** [comment]
@yzslab I have removed `requirements.txt` file as not needed anymore. You can directly add it to `pyproject.toml` directly. I have some suggestions.

- we can create directly cli for some of `utils` scripts with `pyproject.script` mode.
- Let's add `ruff` for consisting code format.

**Review by yzslab — APPROVED** [review]

**Comment by yzslab** [comment]
> @yzslab I have removed `requirements.txt` file as not needed anymore. You can directly add it to `pyproject.toml` directly. I have some suggestions.
> 
> * we can create directly cli for some of `utils` scripts with `pyproject.script` mode.
> * Let's add `ruff` for consisting code format.

Sounds great, but I am busy with other things, so I am not going to do this personally now.
You can do it if you want, as long as it does not change the existing code or files too much. Since I did not write test cases for the most parts of this repo, it takes time to validate the correctness.

Do you want to merge this PR now, or wait untill you add further improvements you mentioned?

**Comment by hardikdava** [comment]
@yzslab 

> Sounds great, but I am busy with other things, so I am not going to do this personally now.
> You can do it if you want, as long as it does not change the existing code or files too much. Since I did not write test cases for the most parts of this repo, it takes time to validate the correctness.

- I will see what I can do with my capabilities. 

> Do you want to merge this PR now, or wait until you add further improvements you mentioned?
- yes, you can merge it now. I will create separate PRs for other things.

**Comment by yzslab** [comment]
Thanks!

**Pull request merged**